specification/servicefabricmanagedclusters/resource-manager/Microsoft.ServiceFabricManagedClusters/stable/2021-05-01/managedcluster.json

```diff
@@ -0,0 +1,1368 @@
+  },
+  "definitions": {
+    "ManagedClusterAddOnFeature": {
+      "type": "string",
+      "description": "Available cluster add-on features",
+      "enum": [
+        "DnsService",
+        "BackupRestoreService",
+        "ResourceMonitorService"
+      ]
+    },
+    "ApplicationTypeVersionsCleanupPolicy": {
+      "required": [
+        "maxUnusedVersionsToKeep"
+      ],
+      "properties": {
+        "maxUnusedVersionsToKeep": {
+          "type": "integer",
+          "format": "int32",
+          "minimum": 0,
+          "description": "Number of unused versions per application type to keep."
+        }
+      },
+      "description": "The policy used to clean up unused versions."
+    },
+    "AzureActiveDirectory": {
+      "properties": {
+        "tenantId": {
+          "type": "string",
+          "description": "Azure active directory tenant id."
+        },
+        "clusterApplication": {
+          "type": "string",
+          "description": "Azure active directory cluster application id."
+        },
+        "clientApplication": {
+          "type": "string",
+          "description": "Azure active directory client application id."
+        }
+      },
+      "description": "The settings to enable AAD authentication on the cluster."
+    },
+    "ClientCertificate": {
+      "required": [
+        "isAdmin"
+      ],
+      "properties": {
+        "isAdmin": {
+          "type": "boolean",
+          "description": "Indicates if the client certificate has admin access to the cluster. Non admin clients can perform only read only operations on the cluster."
+        },
+        "thumbprint": {
+          "type": "string",
+          "description": "Certificate thumbprint."
+        },
+        "commonName": {
+          "type": "string",
+          "description": "Certificate common name."
+        },
+        "issuerThumbprint": {
+          "type": "string",
+          "description": "Issuer thumbprint for the certificate. Only used together with CommonName."
+        }
+      },
+      "description": "Client certificate definition."
+    },
+    "ClusterState": {
+      "type": "string",
+      "description": "The current state of the cluster.\n\n  - WaitingForNodes - Indicates that the cluster resource is created and the resource provider is waiting for Service Fabric VM extension to boot up and report to it.\n  - Deploying - Indicates that the Service Fabric runtime is being installed on the VMs. Cluster resource will be in this state until the cluster boots up and system services are up.\n  - BaselineUpgrade - Indicates that the cluster is upgrading to establishes the cluster version. This upgrade is automatically initiated when the cluster boots up for the first time.\n  - Upgrading - Indicates that the cluster is being upgraded with the user provided configuration.\n  - UpgradeFailed - Indicates that the last upgrade for the cluster has failed.\n  - Ready - Indicates that the cluster is in a stable state.\n",
```

Comment:
>description [](start = 7, length = 11)

The x-ms-enum extension allows you to specify a description for each value of the enumeration. See Microsoft.Blueprint swagger for an example.

Reply:
Actually never mind. See `ClusterUpgradeMode` for an example.

---
In reply to: [612735158](https://github.com/Azure/azure-rest-api-specs/pull/13860#discussion_r612735158) [](ancestors = 612735158)

Reply:
fixed thanks